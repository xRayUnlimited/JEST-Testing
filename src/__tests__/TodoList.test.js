import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConnectedComponent, { TodoList } from '../TodoList';
import { reduxHelper } from '../reduxHelper';
const initialState = { items: [] }

describe(<TodoList />, () => {

  describe('render', () => {
    let component;
    beforeEach( () => {
      const tree = reduxHelper(ConnectedComponent, initialState).component
      component = mount(tree);
    });

    it('matches snapshot', () => {
      expect(toJson(component)).toMatchSnapshot();
    })
  })

  describe('functionality', () => {
    it('updates state on change', () => {
      let component = shallow(<TodoList items={[]} />)
      let input = component.find('input');
      input.simulate('focus');
      input.simulate('change', { target: { name: 'name', value: 'Hello' } })
      expect(component.state('name')).toEqual('Hello');
    })

    it('submits the form', () => {
      let test = { dispatch: jest.fn() }
      let component = mount(<TodoList dispatch={test.dispatch} items={[]} />)
      component.setState({ name: 'Hello' })
      component.find('form').simulate('submit');
      expect(test.dispatch).toHaveBeenCalledTimes(1)
      expect(component.state('name')).toEqual('')
    })


  })

})