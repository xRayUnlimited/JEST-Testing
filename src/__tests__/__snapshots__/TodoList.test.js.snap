// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] render matches snapshot 1`] = `undefined`;

exports[`matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Add Item"
      required={true}
      value=""
    />
  </form>
  <hr />
  <ul />
</div>
`;
